### PR TITLE
[integration] component installer debug, plus pylint stuff, plus better tests

### PR DESCRIPTION
**Pull request by fstagni**

### CODE REVIEW

**Comment by fstagni** [comment]
Can someone review this? I need fixes for Jenkins tests.

File: FrameworkSystem/Client/ComponentInstaller.py [hunk 1]
```diff
@@ -730,6 +736,8 @@ def addDefaultOptionsToCS(self, gConfig_o, componentType, systemName,
       for element in execList:
         result = self.addDefaultOptionsToCS(gConfig_o, componentType, systemName, element, extensions, self.setup,
                                             {}, overwrite)
+        if not result['OK']:
```
Comment (chaen) [inline FrameworkSystem/Client/ComponentInstaller.py]: Is that not a reason to exit ?

File: FrameworkSystem/Client/ComponentInstaller.py [hunk 2]
```diff
@@ -1708,15 +1716,24 @@ def setupSite(self, scriptCfg, cfg=None):
       for system, service in setupServices:
         if not self.addDefaultOptionsToCS(None, 'service', system, service, extensions, overwrite=True)['OK']:
           # If we are not allowed to write to the central CS, add the configuration to the local file
-          self.addDefaultOptionsToComponentCfg('service', system, service, extensions)
+          gLogger.warn("Can't write to central CS, so adding to the specific component CFG")
```
Comment (chaen) [inline FrameworkSystem/Client/ComponentInstaller.py]: maybe specify as variable part the system and service. bellow as well

File: README.rst [hunk 3]
```diff
@@ -37,6 +37,7 @@ DIRAC (Distributed Infrastructure with Remote Agent Control) INTERWARE is a soft
 
 DIRAC has been started by the `LHCb collaboration <https://lhcb.web.cern.ch/lhcb/>`_ who still maintains it. It is now used by several communities (AKA VO=Virtual Organizations) for their distributed computing workflows.
 
+DIRAC is written in python 2.7.13 and transitioning to python 3.
```
Comment (chaen) [inline README.rst]: Maybe it's a good opportunity to add that it currently only run fully on SLC6 ?

File: tests/Jenkins/dirac_ci.sh [hunk 4]
```diff
@@ -368,7 +368,9 @@ function miniInstallDIRAC(){
 
   finalCleanup
 
-  #basic install, with only the CS (and ComponentMonitoring) running, together with DB InstalledComponentsDB, which is needed)
+  killRunsv
```
Comment (chaen) [inline tests/Jenkins/dirac_ci.sh]: why ? Also that will probably not work, as it's ran as `dirac` user, while `runsv` runs as root

Comment (fstagni) [reply tests/Jenkins/dirac_ci.sh]: This is a bad name: it's not killing runsv, but the processes started by runsv. This is just a cleanup procedure.

File: FrameworkSystem/Client/ComponentInstaller.py (hunk 1 above)
Comment (fstagni) [reply FrameworkSystem/Client/ComponentInstaller.py]: I don't know, I didn't want to change the logic here, not for this PR at least.

File: README.rst (hunk 3 above)
Comment (fstagni) [reply README.rst]: This is for integration branch, by that time we'll support also CentOS7 by default.

Comment (chaen) [reply README.rst]: so maybe add the two ? I don't think we say it anywhere

Comment (fstagni) [reply README.rst]: Done.

**Comment by fstagni** [comment]
Please review.

**Comment by chaen** [comment]
there are many comment which are not yours, does it need to be rebased ?

**Comment by fstagni** [comment]
Done.

**Comment by fstagni** [comment]
Review OK now?

**Pull request merged**